src/coreclr/vm/gcenv.ee.cpp

```diff
@@ -49,6 +49,24 @@ VOID GCToEEInterface::SyncBlockCacheWeakPtrScan(HANDLESCANPROC scanProc, uintptr
     SyncBlockCache::GetSyncBlockCache()->GCWeakPtrScan(scanProc, lp1, lp2);
 }
 
+void GCToEEInterface::BeforeGcScanRoots(int condemned, bool is_bgc, bool is_concurrent)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+    }
+    CONTRACTL_END;
+
+#ifdef VERIFY_HEAP
+    // Validate byrefs pinned by IL stubs since the last GC.
+    StubHelpers::ProcessByrefValidationList();
```

Comment:
do you want to only call this when is_bgc is true? this is already called in GcStartWork which means for blocking GCs it would have just been called at the beginning of this GC.

Reply:
Doh. Good call. I will update this to be what it was.